nix/deku.nix

```diff
@@ -0,0 +1,64 @@
+{ pkgs, stdenv, lib, ocamlPackages, doCheck ? true, npmPackages, nodejs ? pkgs.nodejs }:
+
+let 
+  rely = pkgs.ocaml-ng.ocamlPackages_5_00.reason-native.rely.overrideAttrs (_: {
+    postPatch = ''
+      substituteInPlace src/rely/TestSuiteRunner.re --replace "Pervasives" "Stdlib"
+    '';
+  });
+in
+
+ocamlPackages.buildDunePackage {
+  pname = "sidechain";
+  version = "0.0.0-dev";
+
+  src = lib.filterSource {
+    src = ./..;
+    dirs = [ "src" "ppx_let_binding" "tests" ];
+    files = [ "dune-project" "sidechain.opam" "package.json" "package-lock.json" ];
+  };
+
+  configurePhase = ''
+    export NODE_PATH=${npmPackages}/node_modules
+    export PATH=${npmPackages}/node_modules/.bin:$PATH
+
+    ln -s ${npmPackages}/node_modules ./node_modules
```

Comment:
I don't think we need both `export NODE_PATH...` and `ln -s ...`.

For what it's worth, I'm getting `failed to create symbol link `./node_modules/node_modules` whenever node_modules is already tehre.

Reply:
We now do different things in the shell and the build which fixes this. Also remoed the `NODE_PATH`.